### ISSUE TITLE
New rule: ObfuscationRequired

### ISSUE THREAD

**Issue by dgroup** [dgroup/arch4u-pmd#65]
### Overview
Quite often developers writing the logs using statements like 
```java
log.debug("Processing started for customer {}", customer);
// or
log.debug("User {} not authorized for operation {} ", user, operation);
```
As the result, the `toString()` method from `customer` or `user` may have sensitive data that needs to be obfuscated before exposing it to the logs.

The proper fix should be like
```java
log.debug("User {} not authorized for operation {} ", this.obfuscation.apply(user), operation);
```

Related discussion: https://github.com/dgroup/arch4u-pmd/discussions/64.

### New rule requirements
1. Rule should have a property where we may specify `package` or direct class name(s) for objects that should not be passed into `log.debug|info|warn|error|fatal` methods without `obfuscation`
2. Rule should have a property `obfuscation` where we may specify class#method names with expected obfuscation.
3. Rule should throw a violation if the object that has type or package defined in `sensitiveClasses` passed to logger without applying obfuscation from property `obfuscationBy`.

```xml
<rule name="ObfuscationRequired"
        since="0.1.0"
        language="java"
        message="Obfuscation is missing for sensitive logging arguments: https://github.com/dgroup/arch4u-pmd/discussions/64"
        class="io.github.dgroup.arch4u.pmd.ObfuscationRequired">
    ...
    <properties>
      <property name="sensitiveClasses" description="Package(s) or direct class name(s) for objects that should not be passed into logs without obfuscation" value="tbd"/>
      <property name="obfuscationBy" value="ClassA#methodA|com.company.ClassB#methodB|com.company.ClassC" />
    </properties>
  </rule>
```
